student/src/main/java/tds/student/web/handlers/MasterShellHandler.java

```diff
@@ -654,54 +656,12 @@ public boolean evaluate (Object arg0) {
   @ResponseBody
   public ResponseData<TestSummary> scoreTest (HttpServletRequest request) throws ReturnStatusException, TDSSecurityException, StudentContextException {
     checkAuthenticated ();
-
     TestOpportunity testOpp = StudentContext.getTestOpportunity ();
-
-    // validate context
-    if (testOpp == null)
-      StudentContext.throwMissingException ();
-
-    TestManager testManager = new TestManager (testOpp);
-    testManager.LoadResponses (true);
-
-    // If there are more adaptive item groups to take then stop here and
-    // return
-    testManager.CheckIfTestComplete ();
-
-    if (!testManager.IsTestLengthMet ()) {
-      String message = "Review: A student has tried to complete the test but there are still more items left to be generated.";
-      _tdsLogger.applicationError (message, "scoreTest", request, null);
-      HttpContext.getCurrentContext ().getResponse ().setStatus (HttpStatus.SC_FORBIDDEN);
-      return new ResponseData<TestSummary> (TDSReplyCode.Denied.getCode (), message, null);
-    }
-
-    // check if all visible pages are completed
-    PageList pages = testManager.GetVisiblePages ();
-
-    if (!pages.isAllCompleted ()) {
-      String message = "Review: A student has tried to complete the test but all the questions are not completed.";
-      _tdsLogger.applicationError (message, "scoreTest", request, null);
-      HttpContext.getCurrentContext ().getResponse ().setStatus (HttpStatus.SC_FORBIDDEN);
-      return new ResponseData<TestSummary> (TDSReplyCode.Denied.getCode (), message, null);
-    }
-
-    // complete test
-    _oppService.setStatus (testOpp.getOppInstance (), new OpportunityStatusChange (OpportunityStatusType.Completed, true));
-
+    examCompletionService.updateStatusWithValidation(testOpp, new TestManager(testOpp), ExamStatusCode.STATUS_COMPLETED);
+    // Send the exam status to ART
     sendTestStatus (StudentContext.getTestee ().getId (), testOpp.getTestKey (), testOpp.getOppInstance ().getKey (), TestStatusType.COMPLETED);
 
-    // score the test
-    TestScoreStatus scoreStatus = _testScoringService.scoreTest (testOpp.getOppInstance ().getKey (), testOpp.getTestKey ());
-
-    // if we successfully scored the record the server latency
-    if (scoreStatus == TestScoreStatus.Submitted) {
-      ServerLatency latency = ServerLatency.getCurrent (HttpContext.getCurrentContext ());
-      latency.setOperation (ServerLatency.OperationType.Score);
-    }
-
-    // try and get scores
-    return getTestSummary ();
```

Comment:
No need to call `getTestService()` - its a mess of conditionals that ultimately result in the return status below, regardless of settings. isScoredByTDS is always disabled, and there is nothing to do within this method since scoring is not implemented in TDS.